### PR TITLE
Initialize UI

### LINKED ISSUE #2298: Initialize Setup UI project
Setup Typescript + Chakra project

### PR DESCRIPTION
**Pull request by isaack-njama**
Closes #2298 

image

### CODE REVIEW

**Review by elsirion — DISMISSED** [review]
Contains the broken #2308 changes.

**Comment by elsirion** [comment]
You don't have to revert, that leaves an ugly git history, you can instead interactively rebase and drop commits.

**Comment by isaack-njama** [comment]
When I run the server, the [eslint-plugin error](https://drive.google.com/file/d/1PtTfpKRfGMvGjI511HtF7XRjJNpsx46n/view?usp=sharing) keeps showing up.

File: guardian-ui/package.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test"
+  },
+  "lint-staged": {
+    "**/*.{js,jsx,ts,tsx}": [
+      "npx prettier --write",
+      "npx eslint --fix -c ./gateway/ui/.eslintrc.json"
```
Comment (okjodom) [inline guardian-ui/package.json:33]: update to `./guardian-ui/.eslintrc.json`

File: guardian-ui/src/reportWebVitals.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { ReportHandler } from "web-vitals"
+
+const reportWebVitals = (onPerfEntry?: ReportHandler) => {
+  if (onPerfEntry && onPerfEntry instanceof Function) {
+    import("web-vitals").then(({ getCLS, getFID, getFCP, getLCP, getTTFB }) => {
```
Comment (okjodom) [inline guardian-ui/src/reportWebVitals.ts:5]: remove all telemetry reporters. We'll add in only what we need

File: guardian-ui/public/manifest.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "React App",
```
Comment (okjodom) [inline guardian-ui/public/manifest.json:2]: we should customize in this init PR

File: flake.nix [hunk 4]
```diff
@@ -444,7 +444,7 @@
                 ];
               };
 
-              gateway-ui = pkgs.mkShell (shellCommonNative
+              web = pkgs.mkShell (shellCommonNative
```
Comment (okjodom) [inline flake.nix:447]: maybe rename the shell to `fedimint-ui`? we should make sure to update the docs that show how to launch and use this shell

File: guardian-ui/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
+This project was bootstrapped with
```
[the remark this answers is not in the document]
Comment (justinmoon) [reply guardian-ui/README.md:1]: That is what I did in https://github.com/fedimint/fedimint/compare/master...justinmoon:fedimint:guardian-ui

[the remark this answers is not in the document]
Comment (isaack-njama) [reply guardian-ui/README.md:1]: Is there a name for the project? I think we can do better than Guardian UI 😂

[the remark this answers is not in the document]
Comment (okjodom) [reply guardian-ui/README.md]: guardian-ui ;)

File: guardian-ui/public/manifest.json (hunk 3 above)
Comment (okjodom) [reply guardian-ui/public/manifest.json]: excellent

**Comment by jkitman** [comment]
Are we going to import styles from https://www.bitcoinuikit.com/ ?

**Comment by garyray-k** [comment]
Should there be a README for this?

**Pull request merged**